app/demo-loop/KernelUtils.i.hh

```diff
@@ -102,9 +104,9 @@ CELER_FUNCTION void move_and_select_model(const CutoffView&      cutoffs,
 
     // Calculate energy loss over the step length
     auto eloss = calc_energy_loss(cutoffs, mat, particle, phys, step, rng);
-    *edep += eloss.value();
+    *edep += qvalue<Energy>(eloss);
     particle.energy(
-        ParticleTrackView::Energy{particle.energy().value() - eloss.value()});
+        Energy{qvalue<Energy>(particle.energy() - qvalue<Energy>(eloss)});
```

Comment:
```suggestion
        Energy{qvalue<Energy>(particle.energy()) - qvalue<Energy>(eloss)});
```

Reply:
Sorry, clearly I didn't test this on a GPU-enabled system before pushing... pity the CI is down.